draft-ietf-jsonpath-base.md

```diff
@@ -1496,9 +1582,9 @@ filter expression:
 $[?length(@.authors) >= 5]
 ~~~
 
-Its only argument is an optional value (possibly taken from a singular path as
-in the example above).  The result also is an optional value, an unsigned
-integer.
+Its only argument is an instance of `ValueType` (possibly taken from a
+singular path as in the example above).  The result also is an
```

Comment:
We also have to allow the argument to be a function expression, in which case it should be an analog of a Singular Path like `OptionalNodeOrValueType`.

Reply:
The only kind of function that is valid here would be one that returns `ValueType`.  "An instance of `ValueType`" includes this.

The result of a Singular Path is also convertible to `ValueType`, so no issues there.